homeassistant/components/modbus/__init__.py

```diff
@@ -264,6 +265,7 @@
             vol.Optional(CONF_DEVICE_CLASS): SENSOR_DEVICE_CLASSES_SCHEMA,
             vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
             vol.Optional(CONF_REVERSE_ORDER): cv.boolean,
+            vol.Optional(CONF_LAST_RESET): cv.datetime,
```

Comment:
I don't think it should be use configurable like this. IMHO, uses should be able to indicate this is metered value or not and auto-detect the last reset.

Reply:
How would you be able to auto-detect the last reset, in a generic integration like modbus ? I can understand your argument in a device specific integration.

Reply:
See ESPHome as an example, they have implemented such logic, without asking for it.

Reply:
Let me take a look. Btw. ESPHome (last time I looked) was pretty device specific, but that may have changed.

Reply:
@frenck as usual you are correct :smile: that algorithm looks like something I can implement.

Putting this PR in draft status, while preparing a better solution.